### ISSUE TITLE
Lost of "Page Mode" & preset zoom in "Reduce PDF Size" tutorials

### ISSUE THREAD

**Issue by macdeport** [py-pdf/pypdf#1654]
Using https://pypdf.readthedocs.io/en/stable/user/file-size.html examples I experienced lost of "Page Mode" & "preset zoom" in output files

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python3 -m platform
macOS-10.10.5-x86_64-i386-64bit

$ python3 -c "import pypdf;print(pypdf.__version__)"
3.4.1
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader, PdfWriter

dir=''

reader = PdfReader(dir+'bid.pdf')
writer = PdfWriter()

for page in reader.pages:
    page.compress_content_streams()  # This is CPU intensive!
    writer.add_page(page)

writer.add_metadata(reader.metadata)

with open(dir+'out.pdf', "wb") as f:
    writer.write(f)
```

Share here the PDF file(s) that cause the issue. The smaller they are, the
better. Let us know if we may add them to our tests!
Input file 
[input.pdf.zip](https://github.com/py-pdf/pypdf/files/10808515/bid.pdf.zip)
Output file
[out.pdf.zip](https://github.com/py-pdf/pypdf/files/10808550/out.pdf.zip)

OK to use these files for your tests

## Traceback

None

**Comment by pubpub-zz** [comment]
please provide the code you are using

**Comment by macdeport** [comment]
```Python
from pypdf import PdfReader, PdfWriter

dir=''

reader = PdfReader(dir+'bid.pdf')
writer = PdfWriter()

for page in reader.pages:
    page.compress_content_streams()  # This is CPU intensive!
    writer.add_page(page)

writer.add_metadata(reader.metadata)

with open(dir+'out.pdf', "wb") as f:
    writer.write(f)

```

**Comment by pubpub-zz** [comment]
"Page Mode" & "preset zoom" are stored in the Pdf Catalog not in the pages below.
You should wait for #1613 to be merged (soon), then I would recommend this writing:
```
from pypdf import PdfReader, PdfWriter

dir=''

reader = PdfReader(dir+'bid.pdf')
writer = PdfWriter()

writer.clone_document_from_reader(r)
for page in writer.pages:
    page.compress_content_streams()

writer.write(dir+'out.pdf')
```

**Comment by macdeport** [comment]
Thank @pubpub-zz . I have to tune to
```Python
writer.clone_document_from_reader(reader)

```
but lead to:
```Python
Traceback (most recent call last):
  File "...", line 14, in <module>
    page.compress_content_streams()
AttributeError: 'DictionaryObject' object has no attribute 'compress_content_streams'

```

**Comment by pubpub-zz** [comment]
As said the bug you're reporting is already pending to be merged. If you don't want to patch your version wait for a very few days ☺️

**Comment by macdeport** [comment]
@pubpub-zz Thanks. Sorry I misunderstood your clear message. I would wait patiently ;-)

**Comment by MartinThoma** [comment]
I hope  that I can merge #1613 today and release latest tomorrow :-) I'm currently trying to understand what the issue was / how the fix works :sweat_smile:

**Comment by macdeport** [comment]
Maybe useful to know what not to lose...

**Direct OpenAction**:
![bid1-catalog](https://user-images.githubusercontent.com/26068595/221354053-56895513-c786-4048-9864-efcefac9ab91.jpg)

[bid1.pdf.zip](https://github.com/py-pdf/pypdf/files/10830967/bid1.pdf.zip)

**Indirect OpenAction**:
![bid2-catalog](https://user-images.githubusercontent.com/26068595/221354089-06158564-d29c-46f5-966f-98f4f0513928.jpg)
![bid2-openaction](https://user-images.githubusercontent.com/26068595/221354106-7392262f-5ea6-4785-babf-ea418ca4c455.jpg)

[bid2.pdf.zip](https://github.com/py-pdf/pypdf/files/10830966/bid2.pdf.zip)

**Comment by macdeport** [comment]
Using this code I am unsure if version 3.5.0 is expected to fix issue:
 https://github.com/py-pdf/pypdf/issues/1654#issuecomment-1442459566
The output file is now a broken PDF...

**Comment by pubpub-zz** [comment]
post to get the file extracted for test:
[bid1.pdf](https://github.com/py-pdf/pypdf/files/10935632/bid1.pdf)

**Comment by macdeport** [comment]
`bid1.pdf / bid1pubpub.pdf  Adobe Acrobat Pro 10.1.16` properties

If that helps... You are welcome

![bid1-properties](https://user-images.githubusercontent.com/26068595/224158110-df7499a7-cdca-4f9e-b772-e154047faf12.jpg)

![bid1-pubpub-properties](https://user-images.githubusercontent.com/26068595/224158145-c1da15fb-c3bf-495f-9d2e-870e4b942035.jpg)

**Comment by MartinThoma** [comment]
I've just merged #1698 into `main`. Thank you @pubpub-zz :heart: 

It will be released today in `pypdf>3.5.1`
